src/storage/table/column_segment.cpp

```diff
@@ -9,10 +9,66 @@ ColumnSegment::ColumnSegment(TypeId type, ColumnSegmentType segment_type, idx_t
       stats(type, type_size) {
 }
 
+ColumnSegment::ColumnSegment(TypeId type, ColumnSegmentType segment_type, idx_t start, idx_t count,uint64_t* stats_min, uint64_t* stats_max)
+    : SegmentBase(start, count), type(type), type_size(GetTypeIdSize(type)), segment_type(segment_type),
+      stats(type, type_size,stats_min,stats_max) {
+}
+
 SegmentStatistics::SegmentStatistics(TypeId type, idx_t type_size) : type(type), type_size(type_size) {
 	Reset();
 }
 
+template <class T> static void set_min_max(T min_value, T max_value,  T *__restrict min, T *__restrict max) {
+    *min = min_value;
+    *max = max_value;
+}
+
+SegmentStatistics::SegmentStatistics(TypeId type, idx_t type_size, uint64_t* stats_min, uint64_t* stats_max) : type(type), type_size(type_size){
```

Comment:
Could you change the uint64_t* used for stats_min and stats_max to data_t[8]? It is a bit hacky to use a uint64_t type as an 8-byte store, and prone to errors (as if you would put e.g. a double in there in the wrong way, it would lead to rounding/potential overflows).